crates/bevy_gizmos/src/primitives/dim3.rs

```diff
@@ -0,0 +1,822 @@
+
+    // Radius of the cone
+    radius: f32,
+    // Height of the cone
+    height: f32,
+
+    // Center of the base of the cone
+    position: Vec3,
+    // Rotation of the cone
+    //
+    // default orientation is: cone base shape normal is aligned with the `Vec3::Y` axis
+    rotation: Quat,
+    // Color of the cone
+    color: Color,
+
+    segments: usize, // Number of segments used to approximate the cone geometry
+}
+
+impl<T: GizmoConfigGroup> Cone3dBuilder<'_, '_, '_, T> {
+    /// Set the number of segments used to approximate the cone geometry.
+    pub fn segments(mut self, segments: usize) -> Self {
+        self.segments = segments;
+        self
+    }
+}
+
+impl<'w, 's, T: GizmoConfigGroup> GizmoPrimitive3d<Cone> for Gizmos<'w, 's, T> {
+    type Output<'a> = Cone3dBuilder<'a, 'w, 's, T> where Self: 'a;
+
+    fn primitive_3d(
+        &mut self,
+        primitive: Cone,
+        position: Vec3,
+        rotation: Quat,
+        color: Color,
+    ) -> Self::Output<'_> {
+        Cone3dBuilder {
+            gizmos: self,
+            radius: primitive.radius,
+            height: primitive.height,
+            position,
+            rotation,
+            color,
+            segments: DEFAULT_NUMBER_SEGMENTS,
+        }
+    }
+}
+
+impl<T: GizmoConfigGroup> Drop for Cone3dBuilder<'_, '_, '_, T> {
+    fn drop(&mut self) {
+        if !self.gizmos.enabled {
+            return;
+        }
+
+        let Cone3dBuilder {
+            gizmos,
+            radius,
+            height,
+            position,
+            rotation,
+            color,
+            segments,
+        } = self;
+
+        let half_height = *height * 0.5;
+
+        {
+            let positions = (0..=*segments)
+                .map(|frac| frac as f32 / *segments as f32)
+                .map(|percentage| percentage * TAU)
+                .map(|angle| Vec2::from(angle.sin_cos()) * *radius)
+                .map(|p| Vec3::new(p.x, -half_height, p.y))
+                .map(rotate_then_translate_3d(*rotation, *position));
+            gizmos.linestrip(positions, *color);
+        };
+
+        let end = Vec3::Y * half_height;
+        circle_coordinates(*radius, *segments)
+            .map(|p| Vec3::new(p.x, -half_height, p.y))
+            .map(move |p| [p, end])
+            .map(|ps| ps.map(rotate_then_translate_3d(*rotation, *position)))
+            .for_each(|[start, end]| {
+                gizmos.line(start, end, *color);
+            });
+    }
+}
+
+// conical frustum 3d
+
+/// Builder for configuring the drawing options of [`ConicalFrustum3d`].
+pub struct ConicalFrustum3dBuilder<'a, 'w, 's, T: GizmoConfigGroup> {
+    gizmos: &'a mut Gizmos<'w, 's, T>,
+
+    // Radius of the top circle
+    radius_top: f32,
+    // Radius of the bottom circle
+    radius_bottom: f32,
+    // Height of the conical frustum
+    height: f32,
+
+    // Center of the base circle of the conical frustum
+    position: Vec3,
```

Comment:
Outdated; "base" also isn't as clear for conical frusta as it is for cones
```suggestion
    // Center of the conical frustum, half-way between the top and bottom
    position: Vec3,
```